src/scenes/level-selection-scene.ts

```diff
@@ -264,7 +264,12 @@ export class LevelSelectionScreen {
   };
 
   private drawLevel(levelBtn: any, gameLevelData: []) {
-    if (levelBtn.levelData.index + this.levelSelectionPageIndex <= this.data.levels.length) {
+    const currentLevelIndex = levelBtn.levelData.index + this.levelSelectionPageIndex;
+    const currentLevel = this.previousPlayedLevelNumber + 1;
+    
+    levelBtn.setPulse(currentLevelIndex === currentLevel, levelBtn.levelData.isSpecial);
```

Comment:
Can we add this logic to levelBtn.draw?

 if (this.pulse) {   <------ replace this with the currentLevelIndex === currentLevel parameter, 
            this.applyPulseEffect();
        }
        
 I think this way it will me one less state to manage because you won't need to use a true or false flag.

Reply:
thank you @bernhardccodev . I adjusted the code and remove the setpulse function since we dont need it anymore.